daemons/controld/controld_membership.c

```diff
@@ -280,7 +280,7 @@ search_conflicting_node_callback(xmlNode * msg, int call_id, int rc,
         g_hash_table_iter_init(&iter, crm_peer_cache);
         while (g_hash_table_iter_next(&iter, NULL, (gpointer *) &node)) {
             if ((node != NULL)
-                && pcmk__str_eq(node->uuid, node_uuid, pcmk__str_casei)
+                && pcmk__str_eq(node->xml_id, node_uuid, pcmk__str_casei)
                 && pcmk__str_eq(node->name, node_uname, pcmk__str_casei)) {
```

Comment:
I thought that may cause unexpected issues, since we always compare node names case-insensitively and a Pacemaker Remote node's so-called `xml_id` is just its node name.

Reply:
True, and node XML IDs are not the XML ID type, so probably best to leave it

Reply:
Though multiple entries differing only in case would likely cause havoc

Reply:
> havoc

I agree but fixing that should probably come if/when we overhaul node IDs more broadly. I remember it turning into a big mess when we looked into it previously. For uniqueness, we might have to add a UUID suffix (e.g., "node1-status-<generated_uuid>").

Reply:
Yeah more likely we should lowercase all node names for XML ID purposes, and compare case-sensitively.

Making the node XML ID use the XML ID type would be a major pain due to node_state. node_state entries would need a (separate) unique XML ID, but the controller only knows the cluster layer ID, so that would require jumping through hoops. It would not be possible to make node_state compatible with both the current and new IDs, so we'd need to maintain a whole new node state subtree in the new format in addition to maintaining node_state in the old format if the DC needed it.